### PR TITLE
fix: Make code compatible with `miden-base`'s `next`

### PR DESCRIPTION
**Pull request by igamigo**
After refactors/additions of https://github.com/0xPolygonMiden/miden-base/pull/572, compatibility with `miden-base`'s `next` branch was broken. This PR fixes this.

### CODE REVIEW

File: block-producer/src/block_builder/prover/tests.rs [hunk 1]
```diff
@@ -393,6 +394,7 @@ async fn test_compute_note_root_success() {
             note_digest.into(),
             NoteMetadata::new(account_id, NoteType::OffChain, 0.into(), ONE).unwrap(),
         )
+        .unwrap()
```
Comment (mFragaBA) [inline block-producer/src/block_builder/prover/tests.rs:397]: can we use an expect here?

File: block-producer/src/server/api.rs [hunk 2]
```diff
@@ -30,6 +30,9 @@ impl<BB, TV> BlockProducerApi<BB, TV> {
 }
 
 #[tonic::async_trait]
+// FIXME: remove the allow when the upstream clippy issue is fixed:
+// https://github.com/rust-lang/rust-clippy/issues/12281
+#[allow(clippy::blocks_in_conditions)]
```
Comment (mFragaBA) [inline block-producer/src/server/api.rs:35]: is it possible to put it above the `async_trait` attribute like it's done for `state_view/mod.rs`?

Comment (igamigo) [reply block-producer/src/server/api.rs:34]: Done

**Review by hackaugusto — APPROVED** [review]
lgtm

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**